db/commitlog/commitlog.cc

```diff
@@ -114,6 +114,9 @@ db::commitlog::config db::commitlog::config::from_db_config(const db::config& cf
     if (cfg.commitlog_flush_threshold_in_mb() >= 0) {
         c.commitlog_flush_threshold_in_mb = cfg.commitlog_flush_threshold_in_mb();
     }
+    if (!cfg.commitlog_compression().empty()) {
+        c.file_version = descriptor::segment_version_3;
+    }
```

Comment:
Too risky IMO.

Reply:
What do you mean? Obviously not compatible with live update, but could make that part ensure we can not turn _on_ compression from no compression, unless it was turned on at boot. 
Using v3 format adds a uint32 per entry, which, given we alread are chasing commitlog size, seems counter productive...

Reply:
It should be per segment, not per entry.

The reason I dislike it is that it introduces too much variance. It's true that due to downgrades we must support writing old+new, but if we write new exclusively, we can drop old after a while since there won't be any writers for it.

If we keep writing old+new indefinitely, we carry the extra weight forever.

Reply:
Yes. Question is whether to bite the bullet now or later.

I can see a use case for an extended format: allow a mutation to span multiple segments, for large transactions.

I can also see a way to avoid it: have a larger segment size for the group0 commitlog.